src/gatsby-node.js

```diff
@@ -302,6 +305,40 @@ exports.sourceNodes = async (
         );
       }
     };
+    ws.onclose = function() {
+      reconnectingWebSocket(ws, previousDelay);
+    };
+    ws.onerror = function(err) {
+      reporter.error(err.stack);
+    };
+
+    ws.on('pong', () => {
+      count--;
+    });
+
+    const intervalId = setInterval(() => {
```

Comment:
This `intervalid` should be used to cancel next hearbeat at `ws.onerror` and `ws.onclose` the prevent hearbeat ping to be sent on closed connection.